lib/rules/no-implicit-coercion.js

```diff
@@ -300,6 +300,12 @@ module.exports = {
 
                     report(node, recommendation, true);
                 }
+
+                // -(-foo)
+                operatorAllowed = options.allow.includes("- -");
+                if (!operatorAllowed && options.number && node.operator === "-" && node.argument.type === "UnaryExpression" && node.argument.operator === "-" && !isNumeric(node.argument.argument)) {
+                    report(node);
```

Comment:
The message for this error now becomes "use \`undefined\` instead".

I think it's fine to keep the recommendation in the message, we can just make the error non-fixable (the third argument passed to `report()` should be `false`).

Reply:
Oh, apologies, hadn't properly tested my changes

Fixing it

Reply:
Not sure: should I revert b34479565c6998d5d062433f5f9ed8f5fb690e72 and 2a9b163b5a31054482d4353f842d1f7ad3b8f14c too?

Seems like those are autofix tests

Reply:
> Not sure about this: should I revert [b344795](https://github.com/eslint/eslint/commit/b34479565c6998d5d062433f5f9ed8f5fb690e72) and [2a9b163](https://github.com/eslint/eslint/commit/2a9b163b5a31054482d4353f842d1f7ad3b8f14c) too?
> 
> Seemed like those were autofix tests so I removed the one for `- -`

Yes, we should test whether the rule reports an error for `-(-foo)`.

Reply:
No we have those tests, I just removed the one about the autofix behavior